server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableUtils.java

```diff
@@ -441,10 +441,17 @@ private UserCompactionHelper(CompactionConfig compactionConfig, Tablet tablet,
       }
 
       if (selectedFiles.isEmpty()) {
-        tablet.setLastCompactionID(compactionId);
-
-        MetadataTableUtil.updateTabletCompactID(tablet.getExtent(), compactionId,
-            tablet.getTabletServer().getContext(), tablet.getTabletServer().getLock());
+        Boolean updateMetadataTabletFail = false;
+        try {
+          MetadataTableUtil.updateTabletCompactID(tablet.getExtent(), compactionId,
+              tablet.getTabletServer().getContext(), tablet.getTabletServer().getLock());
+        } catch (Exception exc) {
+          updateMetadataTabletFail = true;
+          log.error(exc.getMessage());
+        }
```

Comment:
This catches exceptions that were previously thrown. Is this a valid change in behavior? Why wouldn't it be sufficient to simply reorder these lines, and let the exception be thrown?

Reply:
I see what you mean. updateTabletCompactID does not throw any exception. I thought that this would guarantee no attempts to set the LastCompactionID if the Metadata table update failed. Thinking it over, simply reordering the method calls should work.